extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -223,10 +226,24 @@ export class AmpAnalytics extends AMP.BaseElement {
       .then((services) => {
         this.instrumentation_ = services[0];
         this.variableService_ = services[1];
-        return new AnalyticsConfig(this.element).loadConfig();
+        const loadConfigDeferred = new Deferred();
+        const loadConfigTask = () => {
+          return new AnalyticsConfig(this.element)
+            .loadConfig()
+            .then((config) => {
+              loadConfigDeferred.resolve(config);
```

Comment:
Nit: It's generally better to resolve with a promise, instead of `then`ing and resolving with a value. That way, rejected promise state will also be propagated (if it's possible, not sure here).

So, `configPromise = new AC(…).loadConfig(); loadConfigDeferred.resolve(configPromise)`

Reply:
Good point! Switched to resolve with a promise.